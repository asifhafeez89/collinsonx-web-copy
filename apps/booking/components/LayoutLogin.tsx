import {
  Box,
  Center,
  Container,
  Divider,
  Stack,
} from '@collinsonx/design-system/core';
import React, { ReactNode } from 'react';
import { getThemeKey } from '@lib';

import {
  LogoCergea,
  LogoAmex,
  LogoDinersClubWhite,
} from '@collinsonx/design-system/assets/logo';

interface LayoutProps {
  children: ReactNode;
}

const logos = {
  experienceX: LogoCergea,
  amexBlack: LogoAmex,
  amexPlatinum: LogoAmex,
  dinersClub: LogoDinersClubWhite,
};

const themeKey = getThemeKey();

const Logo = logos[themeKey as keyof typeof logos] ?? LogoCergea;

export default function LayoutLogin({ children }: LayoutProps) {
  return (
    <Container
      pt={10}
      px={0}
      sx={{
        maxWidth: '100%',
        height: '100%',
        overflow: 'hidden',
        backgroundColor: '#F3F2F3',
      }}
    >
      <Box
        sx={{
          borderBottom: '1px solid #cccc',
          width: '100%',
        }}
      >
        <Center pb={8} pt={8} mt={-10} sx={{ backgroundColor: '#ffffff' }}>
          <Logo />
        </Center>
      </Box>
      <Box pt={10} sx={{ margin: '0 auto' }}>
        {children}
      </Box>
    </Container>
  );
}
