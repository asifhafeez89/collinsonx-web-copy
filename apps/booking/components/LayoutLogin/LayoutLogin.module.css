.container {
  max-width: 100%;
  overflow: hidden;
  background-color: var(--background);
}
.container2 {
  border-bottom: 1px solid var(--boxBorder);
  background-color: var(--white);
  width: 100%;
  position: fixed;
  z-index: 200;
}
.container3 {
  width: 100%;
  background-color: var(--white);
  box-shadow: 4px 4px 4px 0px var(--shadow);
  position: fixed;
  z-index: 200;
}
.logoWrapper {
  width: 100%;
  background-color: var(--white);
  box-shadow: 4px 4px 4px 4px var(--shadow);
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  align-items: center;
  height: 100px;
  @media (max-width: 768px) {
    height: 50px;
  }
  z-index: 200;
}

.contentWrapper {
  padding-bottom: 1.3rem;
  margin-top: 6.5rem;
  @media (max-width: 768px) {
    margin-top: 3.3rem;
    padding: 0;
  }
}
