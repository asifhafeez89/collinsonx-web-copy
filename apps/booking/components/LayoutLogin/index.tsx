import { Box, Container, ScrollArea } from '@collinsonx/design-system/core';
import { useViewportSize } from '@collinsonx/design-system/hooks';
import React, { ReactNode } from 'react';

import usePayload from 'hooks/payload';
import AppLogo from '../AppLogo';

import classes from './LayoutLogin.module.css';

interface LayoutProps {
  children: ReactNode;
}

export default function LayoutLogin({ children }: LayoutProps) {
  const { payload, setPayload } = usePayload();
  const { height } = useViewportSize();

  return (
    <ScrollArea type="never">
      <Container
        px={0}
        fluid
        className={classes.container}
        style={{
          minHeight: `${height}px`,
        }}
      >
        <div className={classes.container2}>
          {payload && (
            <Box className={classes.container3}>
              <Box mb={2} mt={2} className={classes.logoWrapper}>
                <AppLogo
                  accountProvider={payload.accountProvider}
                  membershipType={payload.membershipType}
                />
              </Box>
            </Box>
          )}
        </div>
        <Box className={classes.contentWrapper}>{children}</Box>
      </Container>
    </ScrollArea>
  );
}
