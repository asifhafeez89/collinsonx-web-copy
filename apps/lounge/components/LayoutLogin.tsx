import { Container, Stack } from '@mantine/core';

interface LayoutProps {
  children: JSX.Element;
}

export default function LayoutLogin({ children }: LayoutProps) {
  return (
    <Container
      pt={40}
      sx={{
        maxWidth: '375px',
        height: '100%',
        overflow: 'hidden',
      }}
    >
      <Stack spacing={24} sx={{ height: '100%' }}>
        {children}
      </Stack>
    </Container>
  );
}
