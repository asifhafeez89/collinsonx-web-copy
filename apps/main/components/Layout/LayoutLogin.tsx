import { Center, Container, Stack } from '@mantine/core';
import Logo from '../../assets/logo.svg';

interface LayoutProps {
  children: JSX.Element;
}
const Layout = ({ children }: LayoutProps) => {
  return (
    <Container mt={40} sx={{ maxWidth: '375px' }}>
      <Stack spacing={24}>
        <Center>
          <Logo />
        </Center>
        {children}
      </Stack>
    </Container>
  );
};

export default Layout;
