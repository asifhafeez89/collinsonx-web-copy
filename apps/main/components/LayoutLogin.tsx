import { Center, Container, Stack } from '@collinsonx/design-system/core';
import React, { ReactNode } from 'react';
import { getThemeKey } from '@lib';

import {
  LogoExperienceX,
  LogoAmex,
  LogoDinersClubWhite,
} from '@collinsonx/design-system/assets/logo';

interface LayoutProps {
  children: ReactNode;
}

const logos = {
  experienceX: LogoExperienceX,
  amexBlack: LogoAmex,
  amexPlatinum: LogoAmex,
  dinersClub: LogoDinersClubWhite,
};

const themeKey = getThemeKey();

const Logo = logos[themeKey as keyof typeof logos] ?? LogoExperienceX;

export default function LayoutLogin({ children }: LayoutProps) {
  return (
    <Container
      pt={40}
      px={16}
      sx={{
        maxWidth: '375px',
        height: '100%',
        overflow: 'hidden',
        backgroundColor: '#112132',
      }}
    >
      <Stack spacing={40} sx={{ height: '100%' }}>
        <Center pt={8}>
          <Logo />{' '}
        </Center>
        {children}
      </Stack>
    </Container>
  );
}
