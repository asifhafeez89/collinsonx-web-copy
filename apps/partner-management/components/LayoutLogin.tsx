import { LogoCergea } from '@collinsonx/design-system/assets/logo';
import { Flex } from '@collinsonx/design-system/core';
import Link from 'next/link';

interface LayoutLoginProps {
  subHeader?: JSX.Element;
  hasPadding?: boolean;
  children: JSX.Element;
}

export default function LayoutLogin({
  children,
  subHeader,
  hasPadding = true,
}: LayoutLoginProps) {
  return (
    <div
      style={{
        width: '100%',
        height: '100%',
        background: '#f5f5f5',
      }}
    >
      <header
        style={{
          backgroundColor: '#E6E6E8',
          borderBottom: '1px solid #A8A8AA',
          width: '100%',
          padding: '1rem',
          textAlign: 'center',
          margin: 0,
        }}
      >
        <Link href="/" aria-label="Home">
          <LogoCergea />
        </Link>
      </header>
      {subHeader}
      <main
        style={{
          padding: hasPadding ? '32px 40px' : 'auto',
          margin: 0,
          background: '#f5f5f5',
        }}
      >
        <Flex justify="center">{children}</Flex>
      </main>
    </div>
  );
}
