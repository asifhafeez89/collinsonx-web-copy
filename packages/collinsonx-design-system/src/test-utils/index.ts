import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
