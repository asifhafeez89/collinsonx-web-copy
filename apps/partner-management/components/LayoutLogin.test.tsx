import LayoutLogin from './LayoutLogin';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@collinsonx/design-system/test-utils';

describe('<LayoutLogin />', () => {
  it('should render', () => {
    const component = render(
      <LayoutLogin subHeader={<>header</>} hasPadding={true}>
        <></>
      </LayoutLogin>
    );

    expect(component).toMatchSnapshot();
  });
});
