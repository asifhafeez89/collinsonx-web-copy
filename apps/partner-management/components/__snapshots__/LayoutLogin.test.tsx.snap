// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutLogin /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        style="width: 100%; height: 100%; background: rgb(245, 245, 245);"
      >
        <header
          style="background-color: rgb(230, 230, 232); border-bottom: 1px solid #A8A8AA; width: 100%; padding: 1rem; text-align: center; margin: 0px;"
        >
          <a
            aria-label="Home"
            href="/"
          >
            svg
          </a>
        </header>
        header
        <main
          style="padding: 32px 40px; margin: 0px; background: rgb(245, 245, 245);"
        >
          <div
            class="m-8bffd616 mantine-Flex-root __m__-r0"
            style="justify-content: center;"
          />
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      style="width: 100%; height: 100%; background: rgb(245, 245, 245);"
    >
      <header
        style="background-color: rgb(230, 230, 232); border-bottom: 1px solid #A8A8AA; width: 100%; padding: 1rem; text-align: center; margin: 0px;"
      >
        <a
          aria-label="Home"
          href="/"
        >
          svg
        </a>
      </header>
      header
      <main
        style="padding: 32px 40px; margin: 0px; background: rgb(245, 245, 245);"
      >
        <div
          class="m-8bffd616 mantine-Flex-root __m__-r0"
          style="justify-content: center;"
        />
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
